src/app.tsx

```diff
@@ -0,0 +1,779 @@
+  readonly url: string = "https://lh3.googleusercontent.com/u/0/d/1ubPFoTJ1_Ly-Eqn_yMvNhG0Dwq_tuw5c=s4800-k-iv1";
+  readonly width: number = 409;
+  readonly height: number = 636;
+  readonly offset: number = 71;
+  readonly cols: number = 10;
+  readonly n: number = 25;
+}
+
+class ItemUrl {
+  constructor(readonly url: string,
+              readonly numberInPicture: number,
+              readonly n: number) {
+  }
+}
+
+class PersonalGoalProps {
+  readonly kind: string = 'persona-goal-props';
+  readonly offset: number = 510;
+  readonly n: number = 24;
+  readonly divWidth: string = "605px";
+}
+
+class ItemProps {
+  readonly kind: string = 'item-props';
+  readonly url: string;
+  readonly width: number = 292;
+  readonly height: number = 456;
+  readonly cols: number = 10;
+  readonly n: number;
+
+  constructor(private item: ItemUrl, readonly offset: number) {
+    this.item = item;
+    this.url = item.url;
+    this.offset = offset;
+    this.n = item.n;
+  }
+}
+
+type CardRenderProps = RandomItemDesignProps | ItemProps | PersonalGoalProps | RandomSideScenarioProps;
+
+const randomItemDesigns = new RandomItemDesignProps();
+const personalGoals = new PersonalGoalProps();
+
+const itemIdsByProsperityLevel: { [prosperityLevel: number]: Array<number> } = {
+  1: range(1, 14),
+  2: range(15, 7),
+  3: range(22, 7),
+  4: range(29, 7),
+  5: range(36, 7),
+  6: range(43, 7),
+  7: range(50, 7),
+  8: range(57, 7),
+  9: range(64, 7),
+};
+
+const itemUrls = function (): Array<ItemUrl> {
+  const itemCounts = [0, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 4, 4, 4, 2, 2, 2, 2, 2, 4, 2, 2, 2, 2, 2, 2, 4, 2, 2, 2, 2, 2, 2, 4, 2, 2, 2, 2, 2, 2, 4, 2, 2, 2, 2, 2, 2, 4, 2, 2, 2, 2, 2, 2, 4, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1];
+  const itemUrls = [{
+    maxItem: 20,
+    url: "https://lh3.googleusercontent.com/u/0/d/1P0bd7vtA_SVwC7Qm9dJ_YmZkQPNDLOCk=s3200-k-iv2",
+  }, {
+    maxItem: 40,
+    url: "https://lh3.googleusercontent.com/u/0/d/1uHYherEvc9bv3Jpl2TpA2DlRPgGT8l3z=s3200-k-iv2",
+  }, {
+    maxItem: 60,
+    url: "https://lh3.googleusercontent.com/u/0/d/172NPm8x9T8zPE2Vd672_10rZ3ieFHHOX=s3200-k-iv2",
+  }, {
+    maxItem: 90,
+    url: "https://lh3.googleusercontent.com/u/0/d/1XAabPK_Hs8gBXXJVpaBaSsAv6_NioEyu=s3200-k-iv2",
+  }, {
+    maxItem: 133,
+    url: "https://lh3.googleusercontent.com/u/0/d/1KW0TZOs7SDVl5frTM9y-ISR-4dZFIex4=s3200-k-iv2",
+  }, {
+    maxItem: 150,
+    url: "https://lh3.googleusercontent.com/u/0/d/1BHfEwqmC_dax5dV4RFFP5MlfJ76eS7KZ=s3200-k-iv2",
+  }];
+
+  let acc = 0;
+  let itemSheet = 0;
+  return itemCounts.map((num, i) => {
+    if (i >= 71 && i <= 95) {
+      // there are two random items but only one of them is "red", i.e. in this picture
+      num -= 1;
```

Comment:
I can still feel the pain of the night when I wrote this comment ;)

Reply:
:)